apps/src/code-studio/assets/DefaultSpritesEditor.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import {
+  getDefaultList,
+  updateDefaultList
+} from '@cdo/apps/assetManagement/animationLibraryApi';
+import DefaultSpriteRow from '@cdo/apps/code-studio/assets/DefaultSpriteRow';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
+import Button from '@cdo/apps/templates/Button';
+
+export default class DefaultSpritesEditor extends React.Component {
+  state = {
+    isLoading: true,
+    defaultList: {}, // Dictionary with name as key and sprite object as value
+    pendingChangesCount: 0,
+    isUpdating: false,
+    errorText: ''
+  };
+
+  componentDidMount() {
+    getDefaultList()
+      .then(spriteDefault => {
+        let spriteList = {};
+        spriteDefault['default_sprites'].map(
+          sprite => (spriteList[sprite.name] = sprite)
+        );
+        this.setState({defaultList: spriteList, isLoading: false});
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  deleteSpriteFromDefaults = spriteName => {
+    let updatedList = this.state.defaultList;
+    delete updatedList[spriteName];
```

Comment:
this will still mutate `this.state.defaultList` -- in order to make this immutable, we have to make a copy of `this.state.defaultList` rather than assign `updatedList` to its value (`let updatedList = {...this.state.defaultList}`)